packages/api/src/controllers/customAnimalBreedController.js

```diff
@@ -44,6 +46,70 @@ const customAnimalBreedController = {
       }
     };
   },
+
+  addCustomAnimalBreed() {
+    return async (req, res) => {
+      const trx = await transaction.start(Model.knex());
+      try {
+        const { farm_id } = req.headers;
+        let { breed } = req.body;
+        const { default_type_id, custom_type_id } = req.body;
+        breed = baseController.checkAndTrimString(breed);
+
+        if (!breed) {
+          await trx.rollback();
+          return res.status(400).send('Animal breed must be provided');
+        }
+        if (default_type_id && custom_type_id) {
+          await trx.rollback();
+          return res.status(400).send('Only default_type_id or custom_type_id should be specified');
+        }
+        if (!default_type_id && !custom_type_id) {
+          await trx.rollback();
+          return res
+            .status(400)
+            .send('One of default_type_id or custom_type_id should be specified');
+        }
+
+        let breedDetails;
+        if (custom_type_id) {
+          breedDetails = { custom_type_id, breed };
+        } else {
+          breedDetails = { default_type_id, breed };
+        }
+
+        const record = await baseController.existsInTable(trx, CustomAnimalBreedModel, {
+          ...breedDetails,
+          farm_id,
+          deleted: false,
+        });
+
+        if (record) {
+          await trx.rollback();
+          return res.status(409).send();
```

Comment:
What are you thinking -- an error message or the existing record?

For error message I think 409 is enough if we had a consistent way to interpret on frontend.

For existing record I don't think it is a good idea because it is an "addAnimal" endpoint I would expect to only return data if it is added (new). And I think Ivan once mentioned it would be weird to give data for errors as it might help bad actors?

Reply:
`res.sendStatus(409);` and "Conflict" is returned 😄 I learned that it is used for 404 to return "Not found".
I felt that something was wrong with receiving an empty body like this:
<img width="982" alt="Screenshot 2024-01-31 at 8 40 35 AM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/e81ed4fb-d9eb-4caa-8a60-1056d074260f">

However, I also learned that returning an empty body is not uncommon. So that's just my preference, and I'm okay with an empty body!